Gems/Multiplayer/Code/Source/Pipeline/NetworkPrefabProcessor.cpp

```diff
@@ -53,143 +81,115 @@ namespace Multiplayer
+                netEntity->FindComponent<NetBindComponent>()->SetPrefabAssetId(networkSpawnable->GetId());
+                netEntity->FindComponent<NetBindComponent>()->SetPrefabEntityId(prefabEntityId);
             }
             return true;
         });
 
-        instance->GetNestedInstances([&entityToInstanceMap, &netEntities](AZStd::unique_ptr<AzToolsFramework::Prefab::Instance>& nestedInstance)
-        { 
-            GatherNetEntities(nestedInstance.get(), entityToInstanceMap, netEntities);
+        sourceInstance->GetNestedInstances([networkInstance, rootSourceInstance, rootNetworkInstance, prefabName, uniqueName, &context, networkSpawnable](AZStd::unique_ptr<AzToolsFramework::Prefab::Instance>& sourceNestedInstance)
+        {
+            // Make a new nested instance for the network prefab instance
+            AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> networkNestedInstance = AZStd::make_unique<AzToolsFramework::Prefab::Instance>(AzToolsFramework::Prefab::InstanceOptionalReference(*rootNetworkInstance), sourceNestedInstance->GetInstanceAlias());
+            AzToolsFramework::Prefab::Instance& targetNestedInstance = networkInstance->AddInstance(AZStd::move(networkNestedInstance));
+
+            PopulateNetworkInstance(
+                sourceNestedInstance.get(),
+                &targetNestedInstance,
+                rootSourceInstance,
+                rootNetworkInstance,
+                prefabName,
+                uniqueName,
+                context,
+                networkSpawnable);
         });
     }
 
     bool NetworkPrefabProcessor::ProcessPrefab(
         AzToolsFramework::Prefab::PrefabConversionUtils::PrefabProcessorContext& context,
-        AzToolsFramework::Prefab::PrefabConversionUtils::PrefabDocument& prefab,
-        AZ::DataStream::StreamType serializationFormat)
+        AzToolsFramework::Prefab::PrefabConversionUtils::PrefabDocument& prefab)
     {
         using AzToolsFramework::Prefab::PrefabConversionUtils::ProcessedObjectStore;
         using namespace AzToolsFramework::Prefab;
 
         AZStd::string uniqueName = prefab.GetName();
+        AZStd::string prefabName = prefab.GetName();
         uniqueName += NetworkSpawnableFileExtension;
+        prefabName += NetworkFileExtension;
+
+        PrefabConversionUtils::PrefabDocument networkPrefab(prefabName, prefab.GetInstance().GetInstanceAlias());
 
-        auto serializer = [serializationFormat](AZStd::vector<uint8_t>& output, const ProcessedObjectStore& object) -> bool {
+        auto serializer = [](AZStd::vector<uint8_t>& output, const ProcessedObjectStore& object) -> bool {
             AZ::IO::ByteContainerStream stream(&output);
             auto& asset = object.GetAsset();
-            return AZ::Utils::SaveObjectToStream(stream, serializationFormat, &asset, asset.GetType());
+            return AZ::Utils::SaveObjectToStream(stream, AZ::ObjectStream::ST_BINARY, &asset, asset.GetType());
         };
```

Comment:
Why is the network prefab processor still writing out a spawnable by itself instead of letting it pass down the stack and have the prefab catchment processor create the spawnable? It does seem like that a network prefab is added below, so does that mean they overwrite eachother?

Reply:
That might be a phase 2 change. Looping in @SergeyAMZN (author) for more information.

Reply:
The spawnable is written out by the prefab catchment processor indeed. 

It seems this is some leftover code from the previous system, we can get rid of it.

The only reason why it's here is to generate the Asset ID for the network spawnable. 
The asset ID is used later in the `PopulateNetworkInstance` to set it to `NetBindComponents`.
So if we can generate it without actually creating the spawnable, that would work.

Reply:
Investigating...

Reply:
Fixed! Thank you :)